src/index.js

```diff
@@ -249,13 +249,17 @@ Multiaddr.prototype.decapsulate = function decapsulate (addr) {
  * }
  */
 Multiaddr.prototype.getPeerId = function getPeerId () {
-  let b58str = this.stringTuples().filter((tuple) => {
-    if (tuple[0] === protocols.names['ipfs'].code) {
-      return true
-    }
-  })[0][1]
+  let b58str = null
+  try {
+    b58str = this.stringTuples().filter((tuple) => {
+      if (tuple[0] === protocols.names['ipfs'].code) {
+        return true
+      }
+    })[0][1]
+
+    bs58.decode(b58str)
+  } catch (e) {}
 
-  bs58.decode(b58str)
   return b58str
```

Comment:
@dryajov this way you are silencing the error and returning null. Why do you think this is a better approach?